### DISCUSSION TITLE
deployment's build step did not complete within the maximum of 45min

### DISCUSSION THREAD

**Discussion by karanveersingh5623** [vercel/community#5259] [Help]
### Summary

Build keep running for more than 45 minutes , here is the link , I tried all options with build cache and all , not working .
Please suggest a solution or where i can see some errors why building is not completing on time .

There are no errors in the build , why build is taking such a longtime

### Example

https://krs-ranking-email-server-qgmokiae2-karanvs-projects.vercel.app/

### Steps to Reproduce

Please check the vercel link shared above and all the source code and steps are listed in that

**Comment by karanveersingh5623** [comment]
@swarnava  , please help me in this build issue  , let me know if you need more information about the project building

**Comment by amyegan** [comment]
Hey @karanveersingh5623. Thanks for sharing the example deployment link. I checked the logs for that deployment and saw that the install command is set to `npm install && npm start`, so the build never finished and it ran into that 45 minute timeout error. 

It looks like the project may be using Docusaurus. If that's the case, I recommend using the Docusaurus preset for the project's Build & Development Settings. Please give that a try 🙏

**Comment by ajeetraina** [comment]
@amyegan We tried choosing Docusaurus v2 but it says "Docusaurus: not found". 
<img width="1204" alt="image" src="https://github.com/vercel/community/assets/313480/1c617fb6-e8ab-4e99-8e32-1a47bf428fbf">

**Comment by ajeetraina** [comment]
I get this error when I choose:
- Docusaurus V2
- Default Build Commands

<img width="972" alt="image" src="https://github.com/vercel/community/assets/313480/8eb8aee3-6e03-4350-abcb-57258112f6c2">



<img width="1057" alt="image" src="https://github.com/vercel/community/assets/313480/effe1ca7-17ad-47ef-914a-3be227ce466a">

**Comment by karanveersingh5623** [comment]
@amyegan , how can we get this fixed , any update ?

**Comment by amyegan** [comment]
It sounds like the project is missing a dependency. To verify, I just deployed a new project from the [Docusaurus 2](https://vercel.com/templates/react/docusaurus-2) template with any errors. The project uses the `Docusaurus (v2)` preset, so it seem like the preset is working.

Please check that all necessary packages are included in the project's `package.json` file, try deploying again, and let me know if that works for you.

**Comment by ajeetraina** [comment]
May I know what steps you follow? If you can share the screenshot of your package.json, vercel.json and environment variable or framework preset, that would be great.

**Reply by amyegan** [reply]
I used the [Docusaurus 2](https://vercel.com/templates/react/docusaurus-2) template. When deploying from the template, the `Docusaurus (v2)` preset was automatically selected for me. I did not change any of the default or automatic settings.

The code can be found publicly at https://github.com/vercel/vercel/tree/main/examples/docusaurus-2

**Comment by ajeetraina** [comment]
Can you try https://github.com/pingcap/ossinsight? This is the project that
we are trying to deploy and based on Docusaurus v2.

On Wed, Jan 10, 2024 at 7:55 PM Amy ***@***.***> wrote:

> I used the Docusaurus 2 <https://vercel.com/templates/react/docusaurus-2>
> template. When deploying from the template, the Docusaurus (v2) preset
> was automatically selected for me. I did not change any of the default or
> automatic settings.
>
> The code can be found publicly at
> https://github.com/vercel/vercel/tree/main/examples/docusaurus-2
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/vercel/discussions/5259#discussioncomment-8080723>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AACMRCDBQYZVRYA4UML47NLYN2QHNAVCNFSM6AAAAABA6UHEKKVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DAOBQG4ZDG>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 


 ”It is not possible to rescue everyone who is caught in the Windows
quicksand
           --Make sure you are on solid Linux ground before trying.”
